legate/core/partition.py

```diff
@@ -118,55 +131,59 @@ def __eq__(self, other):
         )
 
     @property
-    def runtime(self):
+    def runtime(self) -> Runtime:
         return self._runtime
 
     @property
-    def tile_shape(self):
+    def tile_shape(self) -> Shape:
         return self._tile_shape
 
     @property
-    def color_shape(self):
+    def color_shape(self) -> Shape:
         return self._color_shape
 
     @property
-    def even(self):
+    def even(self) -> bool:
         return True
 
     @property
-    def requirement(self):
+    def requirement(self) -> Type[Partition]:
         return Partition
 
     @property
-    def offset(self):
+    def offset(self) -> Shape:
         return self._offset
 
-    def __hash__(self):
-        if self._hash is None:
-            self._hash = hash(
-                (
-                    self.__class__,
-                    self._tile_shape,
-                    self._color_shape,
-                    self._offset,
-                )
+    def __hash__(self) -> int:
+        if self._hash is not None:
+            return self._hash
+
+        self._hash = hash(
+            (
+                self.__class__,
+                self._tile_shape,
+                self._color_shape,
+                self._offset,
```

Comment:
This could probably be reverted but all things considered I prefer the early return / less indentation

Reply:
Sounds good to me.